components/builder-depot/src/server.rs

```diff
@@ -130,7 +137,12 @@ pub fn origin_create(req: &mut Request) -> IronResult<Response> {
     }
 
     match route_message::<OriginCreate, Origin>(req, &request) {
-        Ok(origin) => Ok(render_json(status::Created, &origin)),
+        Ok(origin) => {
+            // TED: If this fails we need to delete the origin and report an origin create failure
+            // currently there is no mechanism for deleting an origin
+            generate_origin_encryption_keys(origin.get_name(), req)?;
```

Comment:
Hmm, yeah this is a bit tricky if there's no way for a user to recover.  
A secondary question is, how will we seed the already existing origins?

Reply:
If we are also seeding during the first download, then perhaps this generate call should be removed from here..

Reply:
Yeah it's a bit of a pickle. I'm happy to write an internal only origin delete but I was afraid to do it as part of this PR. To answer the secondary question: If a user requests to upload a key and their origin doesn't have an existing encryption key, we generate one and the cli downloads it, then continues on as normal.

Reply:
OK yeah.. is there any use case that would not be satisfied if we just removed this generate call from here then?

Reply:
It's probably safe to remove here